#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("inscribed_radius", double_t, 0, "Defines the inscribed radius.", 0.25, 0.01, 1.0)
gen.add("inflation_radius", double_t, 0, "Defines the maximum inflation radius.", 0.4, 0.01, 3.0)
gen.add("layer_factor", double_t, 0, "The factor to weight this layer", 1.0, 0, 1.0)
gen.add("min_contour_size", int_t, 0, "Defines the minimum size for a contour to be classified as 'lethal'.", 3, 0,
        100000)
gen.add("lethal_value", double_t, 0, "Defines the 'lethal' value for obstacles. -1 results in infinity", 2.0, -1,
        100000)
gen.add("inscribed_value", double_t, 0, "Defines the 'inscribed' value for obstacles.", 1.0, 0, 100000)
gen.add("repulsive_field", bool_t, 0, "Enable the repulsive vector field.", True)
exit(gen.generate("mesh_layers", "mesh_layers", "InflationLayer"))
